### ISSUE TITLE
Removing different actions from different turns/rivers

### ISSUE THREAD

**Issue by bkushigian** [b-inary/postflop-solver#14]
I'm building a strategy simplifier that removes unused sizes from different turns/rivers. For instance, in a single raised pot Bn vs BB on J93 after a small cbet, on a brick turn I'll probably want to overbet barrel, but this size won't be used much on a J or a 9. Therefore I'd like to prune my gametree differently for different turn cards.

It looks like [`ActionTree` doesn't support this](https://github.com/b-inary/postflop-solver/blob/41cb1a301be2037917208a6cdae141f3f89c22f7/src/action_tree.rs#L242-L247):
```
    /// Removes a given line from the action tree.
    ///
    /// - `line` must exist in the current tree.
    /// - Chance actions (i.e., dealing turn and river cards) must be omitted from the `line`.
    /// - If the current node is removed by this method, the current node is moved to the nearest
    ///   ancestor node that is not removed.
```

Is there any way to get this functionality? I also looked into modifying a `PostFlopGame` instance directly but couldn't find anything, and I'd imagine there might be issues that arise from treating isomorphic lines differently?

Thanks a bunch!

**Comment by b-inary** [comment]
Currently, we have not implemented such a feature. I think it is a niche demand to be able to set different action lines for different turn/river cards, so I am afraid I am not very positive about implementing this feature. As a compromise, it might be possible to allow only the deletion of nodes in the `PostFlopGame` struct after building the tree and before allocating memory.

The processing of isomorphisms is also relevant. For example, suppose the flop was Js9h3h. Then, the process for the turn being Jd is also automatically applied to Jc. This is reasonable but somewhat not intuitive.

**Comment by bkushigian** [comment]
That would work for me. I have a bunch of lines I want to remove (which include `chance` actions), and I could manually maintain any isomorphism invariants.

**Comment by b-inary** [comment]
Understood. I will try to address this when I can afford it, but if you need this feature immediately, please work on your fork repository. You only need to reverse the process of `build_tree_recursive`, `push_chances`, and `push_actions` methods and then update `misc_memory_usage`, `num_storage_actions`, and `num_storage_chances` fields.

> (which include `chance` actions)

Are you want to remove chance actions? It is not possible to solve, say, a single specific turn card, because that game tree means the specific turn card is **determined to be dealt**.

**Comment by bkushigian** [comment]
Great, thanks! I'll try to get it implemented on my fork.

**Comment by bkushigian** [comment]
> Are you want to remove chance actions?

No, just responses to them. Adding dead cards might be interesting, but not what I'm concerned with at the moment

**Comment by bkushigian** [comment]
Closed by #16
